script/reference-device/dhcp6_pd_daemon.py

```diff
@@ -0,0 +1,123 @@
+        try:
+            subprocess.run(["sudo", "cp", DHCP_CONFIG_WITH_PD_FILE, DHCP_CONFIG_FILE], check=True)
+            subprocess.run(["sudo", "systemctl", "daemon-reload"], check=True)
+            subprocess.run(["sudo", "service", "dhcpcd", "restart"], check=True)
+            logging.info("Successfully restarted dhcpcd service with pd enabled.")
+            intended_dhcp6pd_state = None
+        except subprocess.CalledProcessError as e:
+            logging.error(f"Error restarting dhcpcd service: {e}")
+            intended_dhcp6pd_state = None
+    else:
+        logging.error(f"{DHCP_CONFIG_WITH_PD_FILE} not found. Cannot apply configuration with pd enabled.")
+        intended_dhcp6pd_state = None
+
+def restore_default_config():
+    if os.path.isfile(DHCP_CONFIG_NO_PD_FILE):
+        try:
+            subprocess.run(["sudo", "cp", DHCP_CONFIG_NO_PD_FILE, DHCP_CONFIG_FILE], check=True)
+            subprocess.run(["sudo", "systemctl", "daemon-reload"], check=True)
+            subprocess.run(["sudo", "service", "dhcpcd", "restart"], check=True)
+            logging.info("Successfully restarted dhcpcd service with pd disabled.")
+        except subprocess.CalledProcessError as e:
+            logging.error(f"Error restarting dhcpcd service: {e}")
+    else:
+        logging.error(f"{DHCP_CONFIG_NO_PD_FILE} not found. Cannot apply configuration with pd disabled.")
+
+def properties_changed_handler(interface_name, changed_properties, invalidated_properties):
+    global intended_dhcp6pd_state
+    if "Dhcp6PdState" in changed_properties:
+        new_state = changed_properties["Dhcp6PdState"]
+        logging.info(f"Dhcp6PdState changed to: {new_state}")
+        if new_state == "running" and intended_dhcp6pd_state != "running":
+            intended_dhcp6pd_state = "running"
+            thread = threading.Thread(target=restart_dhcpcd_with_pd_config)
+            thread.start()
+        elif new_state in ("stopped", "disabled") and intended_dhcp6pd_state is None:
+            restore_default_config()
+
+def connect_to_signal():
+    try:
+        dbus_obj = bus.get_object('io.openthread.BorderRouter.wpan0', '/io/openthread/BorderRouter/wpan0')
+        properties_dbus_iface = dbus.Interface(dbus_obj, 'org.freedesktop.DBus.Properties')
+        dbus_obj.connect_to_signal("PropertiesChanged", properties_changed_handler, dbus_interface=properties_dbus_iface.dbus_interface)
+        logging.info("Connected to D-Bus signal.")
+        return dbus_obj, properties_dbus_iface
+    except dbus.DBusException as e:
+        logging.error(f"Error connecting to D-Bus: {e}")
+        return None, None
+
+def check_and_reconnect(dbus_obj, properties_dbus_iface):
+    if dbus_obj is None:
+        dbus_obj, properties_dbus_iface = connect_to_signal()
+
+def main():
+    # Ensure dhcpcd is running in its last known state. In case it was killed or crashed previously, or
+    # radvd is not yet enabled because of inactive network.target
```

Comment:
It's kind of confusing what this means: `radvd is not yet enabled because of inactive network.target`.

Could you elaborate?

Reply:
The dhcpcd and radvd services have a circular dependency.  dhcpcd attempts to start radvd when the system boots with network prefix delegation (PD) enabled. However, radvd depends on network.target, which cannot start until dhcpcd fully initializes. This prevents radvd from starting automatically with dhcpcd. As a workaround, the PD daemon restarts dhcpcd, which then successfully triggers radvd.

Reply:
Please consider elaborate the comment otherwise it's confusing to suddenly mention `radvd` in the comment.